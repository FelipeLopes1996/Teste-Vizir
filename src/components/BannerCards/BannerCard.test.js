import React from "react";
import { render } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import { BannerCards } from ".";

describe("<BannerCards />", () => {
  it("should render title", () => {
    render(<BannerCards />);
    const element = screen.getByText(/Nossos Planos/i);
    expect(element).toBeInTheDocument();
  });
  it("should render content card", () => {
    render(<BannerCards />);
    const element = screen.getAllByTestId("content");
    expect(element).toHaveLength(3);
  });

  it("should render title card", () => {
    render(<BannerCards />);
    expect(screen.getAllByRole("heading")).toHaveLength(4);
  });

  it("should render snapshot", () => {
    const { container } = render(<BannerCards />);
    expect(container).toMatchSnapshot();
  });
});
