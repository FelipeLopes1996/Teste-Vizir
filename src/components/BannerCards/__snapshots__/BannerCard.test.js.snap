// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerCards /> should render snapshot 1`] = `
<div>
  <h1
    class="sc-gsDKAQ fARGrd"
  >
    Nossos Planos
  </h1>
  <div
    class="sc-dkPtRN jaHQef"
  >
    <div
      class="sc-bdvvtL erRkbr"
    >
      <div>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.885.511a1.745 1.745 0 0 1 2.61.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511zM12.5 1a.5.5 0 0 1 .5.5V3h1.5a.5.5 0 0 1 0 1H13v1.5a.5.5 0 0 1-1 0V4h-1.5a.5.5 0 0 1 0-1H12V1.5a.5.5 0 0 1 .5-.5z"
            fill-rule="evenodd"
          />
        </svg>
        <h1>
          Plano Fale + 30
        </h1>
      </div>
      <p
        data-testid="content"
      >
        Fale de graça até os 30 minutos iniciais de sua ligação, e só paga os
                     minutos excedentes. Os minutos excedentes tem um acrescimo de 
                     10% sobre a tarifa normal do minuto
      </p>
    </div>
    <div
      class="sc-bdvvtL erRkbr"
    >
      <div>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.885.511a1.745 1.745 0 0 1 2.61.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511zM12.5 1a.5.5 0 0 1 .5.5V3h1.5a.5.5 0 0 1 0 1H13v1.5a.5.5 0 0 1-1 0V4h-1.5a.5.5 0 0 1 0-1H12V1.5a.5.5 0 0 1 .5-.5z"
            fill-rule="evenodd"
          />
        </svg>
        <h1>
          Plano Fale + 60
        </h1>
      </div>
      <p
        data-testid="content"
      >
        Fale de graça até os 60 minutos iniciais de sua ligação, e só paga os
      minutos excedentes. Os minutos excedentes tem um acrescimo de 
      10% sobre a tarifa normal do minuto
      </p>
    </div>
    <div
      class="sc-bdvvtL erRkbr"
    >
      <div>
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.885.511a1.745 1.745 0 0 1 2.61.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511zM12.5 1a.5.5 0 0 1 .5.5V3h1.5a.5.5 0 0 1 0 1H13v1.5a.5.5 0 0 1-1 0V4h-1.5a.5.5 0 0 1 0-1H12V1.5a.5.5 0 0 1 .5-.5z"
            fill-rule="evenodd"
          />
        </svg>
        <h1>
          Plano Fale + 120
        </h1>
      </div>
      <p
        data-testid="content"
      >
        Fale de graça até os 120 minutos iniciais de sua ligação, e só paga os
      minutos excedentes. Os minutos excedentes tem um acrescimo de 
      10% sobre a tarifa normal do minuto
      </p>
    </div>
  </div>
</div>
`;
